pkg/localstore/sampler.go

```diff
@@ -180,8 +188,28 @@ func (db *DB) ReserveSample(
 		} else {
 			currentMaxAddr = swarm.NewAddress(make([]byte, 32))
 		}
-		if le(item.Bytes(), currentMaxAddr.Bytes()) || len(sampleItems) < sampleSize {
-			insert(item)
+		if le(item.transformedAddress.Bytes(), currentMaxAddr.Bytes()) || len(sampleItems) < sampleSize {
+
+			chunk := swarm.NewChunk(swarm.NewAddress(item.chunkItem.Address), item.chunkItem.Data)
+
+			stamp := postage.NewStamp(item.chunkItem.BatchID, item.chunkItem.Index, item.chunkItem.Timestamp, item.chunkItem.Sig)
+
+			stampData, err := stamp.MarshalBinary()
+			if err != nil {
+				logger.Info("error marshaling stamp for chunk", chunk.Address(), err)
+				continue
+			}
+			_, err = db.validStamp(chunk, stampData)
+			if err == nil {
+				if !soc.Valid(chunk) && !cac.Valid(chunk) {
```

Comment:
in pullsync, we make these checks before we store the chunks.

Reply:
this check intends to catch if the data would be arbitrarily changed (became invalid) later on

Reply:
can you clarify what you mean?

Reply:
any kind of storage layer unreliability that would be caught in every other use of retrieved data (either local or from the network) wouldn't be caught here without this check.
not doing this check here is explicitly trusting the storage layer